addons/web_unsplash/static/src/components/media_dialog/image_selector.js

```diff
@@ -0,0 +1,256 @@
+/** @odoo-module **/
+
+import { patch } from 'web.utils';
+import { MediaDialog, TABS } from '@web_editor/components/media_dialog/media_dialog';
+import { ImageSelector } from '@web_editor/components/media_dialog/image_selector';
+import { useService } from '@web/core/utils/hooks';
+import { uploadService, AUTOCLOSE_DELAY } from '@web_editor/components/upload_progress_toast/upload_service';
+
+const { useState, Component } = owl;
+
+class UnsplashCredentials extends Component {
+    setup() {
+        this.state = useState({
+            key: '',
+            appId: '',
+            hasKeyError: this.props.hasCredentialsError,
+            hasAppIdError: this.props.hasCredentialsError,
+        });
+    }
+
+    submitCredentials() {
+        if (this.state.key === '') {
+            this.state.hasKeyError = true;
+        } else if (this.state.appId === '') {
+            this.state.hasAppIdError = true;
+        } else {
+            this.props.submitCredentials(this.state.key, this.state.appId);
+        }
+    }
+}
+UnsplashCredentials.template = 'web_unsplash.UnsplashCredentials';
+
+class UnsplashError extends Component {}
+UnsplashError.template = 'web_unsplash.UnsplashError';
+UnsplashError.components = {
+    UnsplashCredentials,
+};
+
+patch(ImageSelector.prototype, 'image_selector_unsplash', {
```

Comment:
@younn-o Just a random comment: Here in this unsplash override, I see all methods defined: none in a private section and even none starting with `_` to indicate some form of "private" stuff. Is that an OWL thing or ... ? Or are those really all public methods now (I see the previous implementation had many "private" one). It's just a convention thing for the javascript side anyway but... :)

Reply:
To be honest, I do not really know if there is a convention, I think there are some components using that convention, and some that don't in the existing codebase. I did not put them as I had in mind that components' methods should not really be used outside of the component, except if passed down as a prop to a child (but on the other hand @sdegueldre just showed a better way to implement the `initWithMedia`/`createElements` operations of the media dialog using static methods, so that was a misconception of mine)

Reply:
Yes, in web we no longer use the _ convention for private methods, as references to components should not be made available, pretty much ever, as their lifecycle and lifetime are managed by owl, passing component references around is a memory leak waiting to happen. All methods that are not explicitly passed as a prop or written in the env are private by default. For services, private methods and private state is captured in the closure so that it is truly private. You can keep using the _ convention in website-related modules if you would like obviously, but that was our reasoning.